app/src/main/java/ua/kozlov/madfood/Foods.java

```diff
@@ -0,0 +1,261 @@
+package ua.kozlov.madfood;
+
+import android.content.Context;
+import org.xmlpull.v1.XmlPullParser;
+import org.xmlpull.v1.XmlPullParserException;
+import java.io.IOException;
+import java.util.ArrayList;
+
+public class Foods implements FoodsInterface {
+
+    @Override
+    public ArrayList<String> getFoodsList(final String group_name, final Context context) {
+        XmlPullParser parser = getParser(group_name, context);
+        ArrayList<String> foodsList = setFoodNamesToList(parser);
+        return foodsList;
+    }
+
+    @Override
+    public String getFoodName(final String groupId, final String id, final Context context) {
+        return null;
+    }
+
+    @Override
+    public String getFoodId(final String groupId, final String name, final Context context) {
+        return null;
+    }
+
+    @Override
+    public ArrayList<String> getFoodParameters(final String groupName, final int position, final Context context) {
+        XmlPullParser parser = getParserFacts(groupName, context);
+        ArrayList<String> foodsList = setFoodParametersToList(parser, position);
+        return foodsList;
+    }
+
+    private ArrayList<String> setFoodNamesToList(final XmlPullParser parser) {
+        ArrayList<String> list = new ArrayList<>();
+        try {
+            while (parser.getEventType() != XmlPullParser.END_DOCUMENT) {
+                if (parser.getEventType() == XmlPullParser.START_TAG
+                        && parser.getName().equals("Food")) {
+                    final String foodName = parser.getAttributeValue(1);
+                    list.add(foodName);
+                }
+                parser.next();
+            }
+        } catch (XmlPullParserException e) {
+            e.printStackTrace();
+        } catch (IOException e) {
+            e.printStackTrace();
+        }
+        return list;
+    }
+
+    private ArrayList<String> setFoodParametersToList(final XmlPullParser parser, final int position) {
+        ArrayList<String> list = new ArrayList<>();
+        int count = 0;
+        try {
+            while (parser.getEventType() != XmlPullParser.END_DOCUMENT) {
+                if (parser.getEventType() == XmlPullParser.START_TAG
+                        && parser.getName().equals("Food")) {
+                    if (count == position) {
```

Comment:
@OlehKozlov  as I know, if arg is **final**, it cann't be modified. So in this case you should write **if (position == count) **, because if you miss one "=" , you will assign position to count, and that will a mistake